lib/log_decorator/log_proxy.rb

```diff
@@ -0,0 +1,50 @@
+module LogDecorator
+  class LogProxy
+    attr_accessor :prefix, :logger
+
+    def self.initialize_log_levels # rubocop:disable AbcSize
+      return if @log_levels_initialized
+      @log_levels_initialized = true
+
+      LogDecorator.log_levels.each do |level|
+        define_method(level) do |msg = nil, &blk|
```

Comment:
I think this is mostly existing code from the original implementation. I did remove the dependency on ActiveSupport (definite the delegates directly).